calico.go

```diff
@@ -306,6 +307,33 @@ func cmdDel(args *skel.CmdArgs) error {
 		"Node":         nodename,
 	}).Info("Extracted identifiers")
 
+	calicoClient, err := CreateClient(conf)
+	if err != nil {
+		return err
+	}
+
+	ep := api.WorkloadEndpointMetadata{
+		Name:             args.IfName,
+		Node:             nodename,
+		ActiveInstanceID: args.ContainerID,
+		Orchestrator:     orchestrator,
+		Workload:         workload,
+	}
+
+	wep, err := calicoClient.WorkloadEndpoints().Get(ep)
```

Comment:
@tomdee @gunjan5 

Whose responsibility is it to police the ActiveInstanceID.  If its CNI then I'm worried there is a mighty big timing window here:

-  Old instance queries Endpoint gets ID - looks OK
-  New instance applies updated Endpoint data (updates the ActiveInstanceID)
-  Old instance goes - yup looks fine, delete.

At the moment, the consumer facing API does not include revision info so there is currently no CompareAndSwap.  That was something that I'd added in my libcalico-go refactor.  So there is no way for CNI to protect against this kind of operation.

Unless we can guarantee the delete is *always* called after a successful create/update then I worry this window will be quite large.

If you moved the ActiveInstanceID checking into libcalico-go then you do have the ability to do the CaS operation.